library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java

```diff
@@ -0,0 +1,80 @@
+package com.novoda.downloadmanager;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class CallbackThrottleCreatorTest {
+
+    @Test
+    public void createsTimeThrottle() {
+        CallbackThrottle callbackThrottle = CallbackThrottleCreator.ByTime(TimeUnit.SECONDS, 1)
+                .create();
+
+        assertThat(callbackThrottle).isInstanceOf(CallbackThrottleByTime.class);
+    }
+
+    @Test
+    public void createsProgressThrottle() {
+        CallbackThrottle callbackThrottle = CallbackThrottleCreator.ByProgressIncrease()
+                .create();
+
+        assertThat(callbackThrottle).isInstanceOf(CallbackThrottleByProgressIncrease.class);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCallbackThrottleDoesNotExist() {
+        CallbackThrottleCreator.ByCustomThrottle(null)
+                .create();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotPublic() {
+        CallbackThrottleCreator.ByCustomThrottle(TestNonPublicCustomThrottle.class)
+                .create();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotFound() {
+        CallbackThrottleCreator.ByCustomThrottle(TestNotFoundCustomThrottle.class)
+                .create();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotInstantiable() {
```

Comment:
Are these things we have code to specifically handle? They sound a bit like language features, as in not our code. If we have cases specifically for them then OK.

Reply:
We have cases for these in the code. I still don't fully understand the need to pass a class type and instantiate in the `CallbackThrottleCreator` but it's there with all the associated catches 😄

Reply:
yeah it does seem odd but :man_shrugging: 

Cool, so long as it's testing our stuff 👍